unittests/resources/frontend_checks.py

```diff
@@ -159,6 +159,19 @@ def __init__(self, sleep_time, **kwargs):
         SleepCheck._next_id += 1
 
 
+class RetriesCheck(BaseFrontendCheck):
+
+    def __init__(self, run_to_pass, file_name, **kwargs):
+        super().__init__(type(self).__name__, **kwargs)
+        self.sourcesdir = None
+        self.valid_systems = ['*']
+        self.valid_prog_environs = ['*']
+        self.executable = 'bash'
+        self.executable_opts = ['-c "echo $(($(cat %s) + 1)) > %s "' %
+                                (file_name, file_name)]
```

Comment:
I would use `pre_run` and `post_run` to make it cleaner and also allow numbering to start from `0`:

```python
self.pre_run = ['current_run=$(cat %s)' % filename]
self.executable = ['echo $current_run']
self.post_run = ['((current_run++))', 'echo $current_run > %s' % filename]
```

Reply:
Great idea! Done.

Reply:
(in addition I had to change the sanity check such that it reads again from `stdout` and not from `filename` - that's what you must have had in mind...)